wsclient/src/main/java/org/hpccsystems/ws/client/HPCCWsDFUClient.java

```diff
@@ -509,6 +511,63 @@ private void checkSuperfileLayouts(String[] subfiles, String cluster) throws Exc
         }
     }
 
+    /**
+     * Create super file
+     * @param superfilename
+     * @param subfiles
+     * @return
+     * @throws Exception
+     * @throws ArrayOfEspExceptionWrapper
+     */
+    public String createSuperfile(String superfilename, String[] subfiles) throws Exception, ArrayOfEspExceptionWrapper
+    {
+        verifyStub(); // Throws exception if stub failed
+
+        AddtoSuperfileRequest request = new AddtoSuperfileRequest();
+
+        request.setSuperfile(superfilename);
+        request.setExistingFile(false);
+
+        //wsdfu > 1.15
+        String comadelimitedsubfiles = "";
+        for (int i = 0; i < subfiles.length; i++)
+        {
+            if (i > 0)
+                comadelimitedsubfiles += ", ";
+
+            comadelimitedsubfiles += subfiles[i];
+        }
+        if (!comadelimitedsubfiles.isEmpty())
+            request.setSubfiles(comadelimitedsubfiles);
+
+        //wsdfu pre 1.15
+        EspStringArray subfilesArray = new EspStringArray();
+        if ( subfiles != null && subfiles.length > 0)
+        {
+            subfilesArray.setItem(subfiles);
+            request.setNames(subfilesArray);
+        }
+
+        AddtoSuperfileResponse resp = null;
+
+        try
+        {
+            resp = ((WsDfuStub) stub).addtoSuperfile(request);
+        }
+        catch (RemoteException e)
+        {
+            throw new Exception("HPCCWsDFUClient.createSuperfile(" + superfilename + ", subfiles) encountered RemoteException.", e);
+        }
+        catch (EspSoapFault e)
+        {
+            handleEspSoapFaults(new EspSoapFaultWrapper(e), "Could Not perform createSuperfile");
+        }
+
+        if (resp.getExceptions() != null)
+            handleEspExceptions(new ArrayOfEspExceptionWrapper(resp.getExceptions()), "Could Not perform createSuperfile");
+
+        return resp.getSubfiles();
+    }
```

Comment:
can you have a method that just takes a request wrapper and returns a response wrapper?  This way the existingFIle flag is available. And if more stuff gets added down the road we won't have to change the interface.